conn.go

```diff
@@ -183,6 +183,10 @@ type halfConn struct {
 	// used to save allocating a new buffer for each MAC.
 	inDigestBuf, outDigestBuf []byte
 
+	// The length of the record header. Always 5 for TLS < 1.2 but maybe
+	// 2 in the TLS 1.3 short header variant.
+	recordHeaderLen int
```

Comment:
Go doesn't have initializators, instead the philosophy is to make the zero value useful.

In this case you would have a bool flag here instead of an int, and then do some

```
recordHeaderLen := normalRecordHeaderLen
if hc.shortRecordHeader {
    recordHeaderLen = shortRecordHeaderLen
}
```

If you find yourself writing that snippet more than 2-3 times, and you really hate it, make a `recordHeaderLen() int` method on `halfConn`.

It might seem less flexible, but you already have `== normalRecordHeaderLen` somewhere, so that flexibility is an illusion and it's best removed. (There are only two valid states of this int, it might as well be a bool.)